src/staticanalysis/bot/static_analysis_bot/clang/format.py

```diff
@@ -104,69 +60,68 @@ def run_clang_format(self, filename, revision):
         with open(clang_output_path, 'w') as f:
             f.write(clang_output)
 
-        # Compare output with original file
-        src_lines = [x.rstrip('\n') for x in open(full_path).readlines()]
-        clang_lines = clang_output.split('\n')
-
-        # Build issues from diff of diff !
-        diff = difflib.SequenceMatcher(
-            a=src_lines,
-            b=clang_lines,
-        )
-        issues = [
-            ClangFormatIssue(filename, src_lines, clang_lines, opcode, revision)
-            for opcode in diff.get_opcodes()
-            if opcode[0] in OPCODES
-        ]
+        # Save any changes that `./mach clang-format` found
+        self.diff = client.diff(unified=8).decode('utf-8')
+
+        if len(self.diff) > 0:
+            # Generate a reverse diff for `parsepatch` (in order to get original
+            # line numbers from the dev's patch instead of new line numbers)
+            reverse_diff = client.diff(unified=8, reverse=True).decode('utf-8')
+
+            # List all the lines that were fixed by `./mach clang-format`
+            patch = Patch.parse_patch(reverse_diff, skip_comments=False)
+            assert patch != {}, \
+                'Empty patch'
+
+            # Build `ClangFormatIssue`s
+            issues = []
+            for filename, diff in patch.items():
+                lines = sorted(diff.get('touched', []) + diff.get('added', []))
+
+                # Group consecutive lines together (algorithm by calixte)
+                groups = []
+                group = [lines[0]]
+                for line in lines[1:]:
+                    # If the line is not consecutive with the group, start a new
+                    # group
+                    if line != group[-1] + 1:
+                        groups.append(group)
+                        group = []
+                    group.append(line)
+                # Don't forget to add the last group
+                groups.append(group)
+
+                issues += [
+                    ClangFormatIssue(filename, group[0], len(group), revision)
+                    for group in groups
+                ]
 
         stats.report_issues('clang-format', issues)
         return issues
 
 
 class ClangFormatIssue(Issue):
     '''
-    An issue created by Clang Format tool
+    An issue created by the Clang Format tool
     '''
-    def __init__(self, path, a, b, opcode, revision):
-        self.mode, self.positions = opcode[0], opcode[1:]
-        assert self.mode in OPCODES
-        assert isinstance(self.positions, tuple)
-        assert len(self.positions) == 4
+    def __init__(self, path, line, nb_lines, revision):
```

Comment:
Why don't you use the modification suggested by clang-format as you have it in the diff produced by mach ?

It should be sent to the developper, and even to us instead of simply saying `Warning: Incorrect coding style [clang-format]`

Reply:
This is extra work that I left out of the MVP in order to be able to finish it faster. I agree that it should be done at some point.